### PR TITLE
Add WebApplicationBuilder Extensions, improve .NET 6 compatibility

### PR DESCRIPTION
**Pull request by TimHess**
#764, #814

### CODE REVIEW

**Comment by lvendramini** [comment]
Is .Net 6 supported?  How do we add steeltoe to .net 6 program.cs?

**Comment by TimHess** [comment]
> Is .Net 6 supported? How do we add steeltoe to .net 6 program.cs?

Thanks for the question... This PR primarily adds extensions to use with the new `WebApplicationBuilder`, otherwise `HostBuilder` and `WebHostBuilder` can both still be used in .NET 6. Additionally, it is possible to use most of the other pre-existing Steeltoe setup methods with WebApplication by using them on the `.Configuration` or `.Services` properties of the builder, eg:

```
builder.Configuration.AddConfigServer();
builder.Logging.AddDynamicConsole();
builder.Services.AddAllActuators(builder.Configuration);
builder.Services.ActivateActuatorEndpoints();
```

I believe the only known issue with Steeltoe on .NET 6 (which is addressed in this PR) is that there is potential for recursion with the placeholder config provider.

**Comment by lvendramini** [comment]
> > Is .Net 6 supported? How do we add steeltoe to .net 6 program.cs?
> 
> Thanks for the question... This PR primarily adds extensions to use with the new `WebApplicationBuilder`, otherwise `HostBuilder` and `WebHostBuilder` can both still be used in .NET 6. Additionally, it is possible to use most of the other pre-existing Steeltoe setup methods with WebApplication by using them on the `.Configuration` or `.Services` properties of the builder, eg:
> 
> ```
> builder.Configuration.AddConfigServer();
> builder.Logging.AddDynamicConsole();
> builder.Services.AddAllActuators(builder.Configuration);
> builder.Services.ActivateActuatorEndpoints();
> ```
> 
> I believe the only known issue with Steeltoe on .NET 6 (which is addressed in this PR) is that there is potential for recursion with the placeholder config provider.

In our Program.cs we have
using Steeltoe.Extensions.Configuration.CloudFoundry;

var builder = WebApplication.CreateBuilder(args);

builder.Configuration.**AddConfigServer**();
builder.Logging.**AddDynamicConsole**();
builder.Services.**AddAllActuators**(builder.Configuration);
builder.Services.**ActivateActuatorEndpoints**();

But none of the bolded entries are know.  What else do we need? Thanks!

**Comment by TimHess** [comment]
> > > Is .Net 6 supported? How do we add steeltoe to .net 6 program.cs?
> > 
> > 
> > Thanks for the question... This PR primarily adds extensions to use with the new `WebApplicationBuilder`, otherwise `HostBuilder` and `WebHostBuilder` can both still be used in .NET 6. Additionally, it is possible to use most of the other pre-existing Steeltoe setup methods with WebApplication by using them on the `.Configuration` or `.Services` properties of the builder, eg:
> > ```
> > builder.Configuration.AddConfigServer();
> > builder.Logging.AddDynamicConsole();
> > builder.Services.AddAllActuators(builder.Configuration);
> > builder.Services.ActivateActuatorEndpoints();
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I believe the only known issue with Steeltoe on .NET 6 (which is addressed in this PR) is that there is potential for recursion with the placeholder config provider.
> 
> In our Program.cs we have using Steeltoe.Extensions.Configuration.CloudFoundry;
> 
> var builder = WebApplication.CreateBuilder(args);
> 
> builder.Configuration.**AddConfigServer**(); builder.Logging.**AddDynamicConsole**(); builder.Services.**AddAllActuators**(builder.Configuration); builder.Services.**ActivateActuatorEndpoints**();
> 
> But none of the bolded entries are know. What else do we need? Thanks!

Assuming you've already got the nuget references (EndpointCore and ConfigServerBase or Core for these, other components would be different), adding these usings should work. Visual Studio should be able to find them via `CTRL` + `.` suggestions too
```
using Steeltoe.Extensions.Configuration.ConfigServer;
using Steeltoe.Extensions.Logging;
using Steeltoe.Management.Endpoint;
```

**Comment by lvendramini** [comment]
> > > > Is .Net 6 supported? How do we add steeltoe to .net 6 program.cs?
> > > 
> > > 
> > > Thanks for the question... This PR primarily adds extensions to use with the new `WebApplicationBuilder`, otherwise `HostBuilder` and `WebHostBuilder` can both still be used in .NET 6. Additionally, it is possible to use most of the other pre-existing Steeltoe setup methods with WebApplication by using them on the `.Configuration` or `.Services` properties of the builder, eg:
> > > ```
> > > builder.Configuration.AddConfigServer();
> > > builder.Logging.AddDynamicConsole();
> > > builder.Services.AddAllActuators(builder.Configuration);
> > > builder.Services.ActivateActuatorEndpoints();
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > I believe the only known issue with Steeltoe on .NET 6 (which is addressed in this PR) is that there is potential for recursion with the placeholder config provider.
> > 
> > 
> > In our Program.cs we have using Steeltoe.Extensions.Configuration.CloudFoundry;
> > var builder = WebApplication.CreateBuilder(args);
> > builder.Configuration.**AddConfigServer**(); builder.Logging.**AddDynamicConsole**(); builder.Services.**AddAllActuators**(builder.Configuration); builder.Services.**ActivateActuatorEndpoints**();
> > But none of the bolded entries are know. What else do we need? Thanks!
> 
> Assuming you've already got the nuget references (EndpointCore and ConfigServerBase or Core for these, other components would be different), adding these usings should work. Visual Studio should be able to find them via `CTRL` + `.` suggestions too
> 
> ```
> using Steeltoe.Extensions.Configuration.ConfigServer;
> using Steeltoe.Extensions.Logging;
> using Steeltoe.Management.Endpoint;
> ```

Thanks that worked.  Will that allow us to host our service in Pivotal Cloud Foundry?  So far it does not work...  Before .net 6 we had in the startup 
`      public static IWebHost CreateWebHostBuilder(string[] args) =>
            WebHost.CreateDefaultBuilder(args)
                .UseStartup<Startup>()
                .UseCloudFoundryHosting()
                .UseDefaultServiceProvider(options => options.ValidateScopes = false)
                .ConfigureAppConfiguration((builderContext, configBuilder) =>
                {
                    configBuilder.AddCloudFoundry();
                }).Build();
`

Does the following allow the service to be containerized in PCF?  Or are we missing something?
 builder.Configuration.AddConfigServer();
builder.Logging.AddDynamicConsole();
builder.Services.AddAllActuators(builder.Configuration);
builder.Services.ActivateActuatorEndpoints();

**Comment by TimHess** [comment]
> > > > > Is .Net 6 supported? How do we add steeltoe to .net 6 program.cs?
> > > > 
> > > > 
> > > > Thanks for the question... This PR primarily adds extensions to use with the new `WebApplicationBuilder`, otherwise `HostBuilder` and `WebHostBuilder` can both still be used in .NET 6. Additionally, it is possible to use most of the other pre-existing Steeltoe setup methods with WebApplication by using them on the `.Configuration` or `.Services` properties of the builder, eg:
> > > > ```
> > > > builder.Configuration.AddConfigServer();
> > > > builder.Logging.AddDynamicConsole();
> > > > builder.Services.AddAllActuators(builder.Configuration);
> > > > builder.Services.ActivateActuatorEndpoints();
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > I believe the only known issue with Steeltoe on .NET 6 (which is addressed in this PR) is that there is potential for recursion with the placeholder config provider.
> > > 
> > > 
> > > In our Program.cs we have using Steeltoe.Extensions.Configuration.CloudFoundry;
> > > var builder = WebApplication.CreateBuilder(args);
> > > builder.Configuration.**AddConfigServer**(); builder.Logging.**AddDynamicConsole**(); builder.Services.**AddAllActuators**(builder.Configuration); builder.Services.**ActivateActuatorEndpoints**();
> > > But none of the bolded entries are know. What else do we need? Thanks!
> > 
> > 
> > Assuming you've already got the nuget references (EndpointCore and ConfigServerBase or Core for these, other components would be different), adding these usings should work. Visual Studio should be able to find them via `CTRL` + `.` suggestions too
> > ```
> > using Steeltoe.Extensions.Configuration.ConfigServer;
> > using Steeltoe.Extensions.Logging;
> > using Steeltoe.Management.Endpoint;
> > ```
> 
> Thanks that worked. Will that allow us to host our service in Pivotal Cloud Foundry? So far it does not work... Before .net 6 we had in the startup ` public static IWebHost CreateWebHostBuilder(string[] args) => WebHost.CreateDefaultBuilder(args) .UseStartup<Startup>() .UseCloudFoundryHosting() .UseDefaultServiceProvider(options => options.ValidateScopes = false) .ConfigureAppConfiguration((builderContext, configBuilder) => { configBuilder.AddCloudFoundry(); }).Build();`
> 
> Does the following allow the service to be containerized in PCF? Or are we missing something? builder.Configuration.AddConfigServer(); builder.Logging.AddDynamicConsole(); builder.Services.AddAllActuators(builder.Configuration); builder.Services.ActivateActuatorEndpoints();

Could you please open a separate issue, listing all of the Steeltoe components you're wanting to use and ideally a complete sample?

File: src/Configuration/src/KubernetesBase/KubernetesServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+using Microsoft.Extensions.DependencyInjection;
+using Steeltoe.Common.Kubernetes;
+
+namespace Steeltoe.Extensions.Configuration.Kubernetes
+{
+    public static class KubernetesServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Adds KubernetesApplicationInstanceInfo and ensures runtime loggers are used
```
Comment (hananiel) [inline src/Configuration/src/KubernetesBase/KubernetesServiceCollectionExtensions.cs:13]:
```suggestion
        /// Adds KubernetesApplicationInstanceInfo and ensures startup loggers are replaced by runtime loggers
```

**Review by hananiel — CHANGES_REQUESTED** [review]

**Review by hananiel — APPROVED** [review]

**Review by hananiel — APPROVED** [review]

**Pull request merged**